install.sh

```diff
@@ -0,0 +1,537 @@
+  DISTRO=${DISTRO:-$(distro)}
+
+  case $DISTRO in
+  # macOS uses the standalone installation for now.
+  # Homebrew support is planned. See: https://github.com/coder/coder/issues/1925
+  macos) install_standalone ;;
+  # The .deb and .rpm files are pulled from GitHub.
+  debian) install_deb ;;
+  fedora | opensuse) install_rpm ;;
+  # We don't have GitHub releases that work on Alpine or FreeBSD so we have no
+  # choice but to use npm here.
+  alpine) install_apk ;;
+  # For anything else we'll try to install standalone but fall back to npm if
+  # we don't have releases for the architecture.
+  *)
+    echoh "Unsupported package manager."
+    echoh "Falling back to standalone installation."
+    install_standalone
+    ;;
+  esac
+}
+
+parse_arg() {
+  case "$1" in
+  *=*)
+    # Remove everything after first equal sign.
+    opt="${1%%=*}"
+    # Remove everything before first equal sign.
+    optarg="${1#*=}"
+    if [ ! "$optarg" ] && [ ! "${OPTIONAL-}" ]; then
+      echoerr "$opt requires an argument"
+      echoerr "Run with --help to see usage."
+      exit 1
+    fi
+    echo "$optarg"
+    return
+    ;;
+  esac
+
+  case "${2-}" in
+  "" | -*)
+    if [ ! "${OPTIONAL-}" ]; then
+      echoerr "$1 requires an argument"
+      echoerr "Run with --help to see usage."
+      exit 1
+    fi
+    ;;
+  *)
+    echo "$2"
+    return
+    ;;
+  esac
+}
+
+fetch() {
+  URL="$1"
+  FILE="$2"
+
+  if [ -e "$FILE" ]; then
+    echoh "+ Reusing $FILE"
+    return
+  fi
+
+  sh_c mkdir -p "$CACHE_DIR"
+  sh_c curl \
+    -#fL \
+    -o "$FILE.incomplete" \
+    -C - \
+    "$URL"
+  sh_c mv "$FILE.incomplete" "$FILE"
+}
+
+install_deb() {
+  echoh "Installing v$VERSION of the $ARCH deb package from GitHub."
+  echoh
+
+  fetch "https://github.com/coder/coder/releases/download/v$VERSION/coder_${VERSION}_${OS}_${ARCH}.deb" \
+    "$CACHE_DIR/coder_${VERSION}_$ARCH.deb"
+  sudo_sh_c dpkg -i "$CACHE_DIR/coder_${VERSION}_$ARCH.deb"
+
+  echo_systemd_postinstall deb
+}
+
+install_rpm() {
+  echoh "Installing v$VERSION of the $ARCH rpm package from GitHub."
+  echoh
+
+  fetch "https://github.com/coder/coder/releases/download/v$VERSION/coder_${VERSION}_${OS}_${ARCH}.rpm" \
+    "$CACHE_DIR/coder_${VERSION}_${OS}_${ARCH}.rpm"
+  sudo_sh_c rpm -i "$CACHE_DIR/coder_${VERSION}_${OS}_${ARCH}.rpm"
+
+  echo_systemd_postinstall rpm
+}
+
+install_apk() {
+  echoh "Installing v$VERSION of the $ARCH apk package from GitHub."
+  echoh
+
+  fetch "https://github.com/coder/coder/releases/download/v$VERSION/coder_${VERSION}_${OS}_${ARCH}.apk" \
+    "$CACHE_DIR/coder_${VERSION}_${OS}_${ARCH}.apk"
+  sudo_sh_c apk add --allow-untrusted "$CACHE_DIR/coder_${VERSION}_${OS}_${ARCH}.apk"
+
+  echo_systemd_postinstall apk
+}
+
+install_standalone() {
+  echoh "Installing v$VERSION of the $ARCH release from GitHub."
+  echoh
+
+  # MacOS releases are packaged as .zip
+  case $OS in
+  darwin) STANDALONE_ARCHIVE_FORMAT=zip ;;
+  *) STANDALONE_ARCHIVE_FORMAT=tar.gz ;;
+  esac
```

Comment:
I'm not super happy with this implementation, had to make some adjustments since our OSX releases are published as .zips